### PR TITLE
feat: add tdp-observability

### LINKED ISSUE #228: Integrate tdp-observability to tdp-getting-started
The [tdp-observability](https://github.com/TOSIT-IO/tdp-observability) repository has been open-sourced by [Alliage](https://www.alliage.io). It has to be integrated into `tdp-getting-started`, notably into the `setup.sh` with a new enable option `-e observability`.

### PR DESCRIPTION
**Pull request by leopaul36**
#### Which issue(s) this PR fixes

Fixes #228 

#### Additional comments

Marking as draft waiting for https://github.com/TOSIT-IO/tdp-observability/pull/20 to be merged. I will update the submodule when it is done.

#### Agreements

- [X] I hereby declare this contribution to be licensed under the [Apache License Version 2.0, January 2004](http://www.apache.org/licenses/LICENSE-2.0).
- [X] I hereby agree to grant [TOSIT](https://www.tosit.io/) a copyright license to use my contributions.

### CODE REVIEW

**Comment by leopaul36** [comment]
Ready to merge since https://github.com/TOSIT-IO/tdp-observability/pull/20 is done

**Comment by rpignolet** [comment]
I think the use of `setup.sh -r latest` will not work because the git checkout is hardcoded to `origin/master` here https://github.com/TOSIT-IO/tdp-getting-started/blob/8fd6507eb7d87c487670e77206648045ec1de844/scripts/setup.sh#L88 and `tdp-observability` use `main` instead of `master`.

**Comment by rpignolet** [comment]
We can add a variable to `git_submodule_setup` function with the branch name for the `latest` case and use it like this `git_submodule_setup "$submodule_path" main`

**Comment by GuillaumeHold** [comment]
The naming "main" is supposed to be the new standard (according to GitHub).
I would rather have a consistent naming with TOSIT standard.

**Comment by rpignolet** [comment]
https://github.com/TOSIT-IO/tdp-getting-started/blob/8fd6507eb7d87c487670e77206648045ec1de844/scripts/setup.sh#L151

I think you should add the new collection in order to generate the `.venv` with the new collection like this:
```
observability) collection_path="${collection_path}:${TDP_COLLECTION_OBSERVABILITY_PATH}" ;; 
```

File: README.md [hunk 1]
```diff
@@ -40,7 +40,7 @@ git clone https://github.com/TOSIT-IO/tdp-getting-started.git
 # Move into project dir
 cd tdp-getting-started
 # Setup local env with stable tdp-collection (mandatory), tdp-lib (mandatory), tdp-server, tdp-collection-extras, tdp-collection-prerequisites, and vagrant
-./scripts/setup.sh -e server -e extras -e prerequisites -e vagrant
+./scripts/setup.sh -e observability -e server -e extras -e prerequisites -e vagrant
```
Comment (rpignolet) [inline README.md:43]: The order is important for collection, `observability` should be after `extras` because it will observe service is extra too no ?

Comment (leopaul36) [reply README.md:43]: Since it's only about the checkout of the submodules in the case of the collections does it really matter?

Comment (rpignolet) [reply README.md:43]: It matter when the script write the `TDP_COLLECTION_PATH` variable inside `.env` file.

Comment (leopaul36) [reply README.md]: Re ordered in the latest push force

Comment (leopaul36) [reply README.md]: Reordered in the latest push force

**Comment by leopaul36** [comment]
> The naming "main" is supposed to be the new standard (according to GitHub). I would rather have a consistent naming with TOSIT standard.

Done in the latest commit

**Comment by leopaul36** [comment]
> `observability) collection_path="${collection_path}:${TDP_COLLECTION_OBSERVABILITY_PATH}" ;; `

Done

**Review by rpignolet — APPROVED** [review]

**Pull request merged**